BoxStation: Added preloaded drone shell dispenser to Testing Lab

coiax opened the pull request:
Drone shells will now be present roundstart, regardless if R&D are
doing their job or not.

There are drones already on the ~~derelicht~~ derelict, which can easily
make their way to the mainstation anyway.

:cl: coiax
rscadd: Boxstation, not to be outdone by all the other competing station designs, has installed a drone shell dispenser in the Testing Lab, which makes automated drones to help fix the station.
/:cl:

coiax committed:
Added preloaded drone shell dispenser to Electrical Maintenance

Drone shells will now be present roundstart, regardless if R&D are
doing their job or not.

There are drones already on the derelicht, which can easily
make their way to the mainstation anyway.

:cl: coiax
tweak: A dusty machine in maintenance has recently whirred to life, making automated drones to help fix the station.
/:cl:

Bawhoppen commented:
:-1: we've had this discussion before and it has ultimately failed not to mention this is bad placement for it if you were going to do it

coiax commented:
I'm reopening the discussion, dammit.

And where would you put it?

Bawhoppen commented:
misc testing lab right above the firealarm remove the locker

coiax committed:
Moved the dispenser to Testing Lab

Based on feedback, moved the dispenser to the Testing Lab, because it's
"full of junk".

Core0verload commented:
>I don't like it. Ghost roles mustn't be guaranteed.

Bawhoppen commented:
@Core0verload drones aren't on the same par as other ghost roles however, they're a minor side thing where only a handful of people enjoy that kind of experience
besides, ghost roles will be basically guaranteed after space ruins stuff is finished

coiax commented:
There are guaranteed ghost roles on the Derelict right now. At any point in a round, if one's left, you can take a shell, go to mainstation, do R&D, print drone shells. Probably take about ten minutes tops. But that's just jumping through a lot of pointless hoops to make drone shells for the station. The only difference between a drone shell dispenser and doing the current solution is that the current solution takes less metal and glass and has like a ten minute setup time.

jughu commented:
DRONES WASTING MUH METAL

ChangelingRain commented on _maps/map_files/TgStation/tgstation.2.1.3.dmm:
```diff
@@ -45408,7 +45408,6 @@
 	pixel_y = -24
 	},
-/obj/structure/closet,
 /obj/item/pipe{
```
Should probably remove this too.

coiax replied:
Remove what?

ChangelingRain replied:
The pipe item.

Cheridan commented:
Roundstart drones sounds fine.

Roundstart golems are just bad because it's removing randomization from a feature based heavily around it for the purpose of maintaining novelty.

coiax commented:
It's technically two seperate PRs, Cheridan.

It wasn't actually planned to do these together, I've always wanted to put a drone shell dispenser on the map, and Free Golems happened during the map freeze.

coiax commented:
Are there any objections to this aside from objecting to guaranteed  ghost roles?

Three drone shells are located on the derelict in constant places, which is constant regardless of main station map. As I said before, the time to set up and print shells on the main station is about 10 minutes.

Here's a thought, what if we put the dispenser there, but set the initial cooldown to say, ten minutes? Then the drones don't appear immediately, but at a predictable time into the round.

TrustyGun commented:
the rest of em merged already pls merge

RandomMarine commented:
This should be testmerged for a while at least. See how things go with roundstart drones. If things go well someone can map out a better spot for the dispenser.

coiax commented:
@RandomMarine: The current spot was suggested by @Bawhoppen, have you got a better one in mind?

RandomMarine commented:
There's tons of space in maintenance that would work, for instance, set up a room in that space gap in engineering maintenance.

Bawhoppen commented:
@RandomMarine no, everyone's thought immediately drifts to being in maint, but thats just a plain bad idea, its the best if its actually in a real place where people can refill it

TrustyGun commented:
i think it may work best here:
![untitled](https://cloud.githubusercontent.com/assets/7079721/14420549/b48e7a52-ff9a-11e5-929f-6456391aa6f9.png)

Bawhoppen commented:
I will fight to the death for it not to be in maint. Putting it in maint is autistic

TrustyGun commented:
Why not place it in maint?

coiax commented:
> the rest of em were merged already pls merge

GunHog commented:
Why do you care so much where it is? Misc. Testing or Maint, it does not seem to matter much, just merge it as is, assuming everything is tested and working.

coiax commented:
Bawhoppen literally picked the location where it is, he has no objections.

GunHog commented:
I have no problem with it being there. Merge it!

coiax committed:
Moved gas mixer fitting to floor nearby

Now instead of in a closet, it's just discarded on a floor.

coiax committed:
Merge branch 'master' into drones-are-the-heros-we-need

Conflicts:
	_maps/map_files/TgStation/tgstation.2.1.3.dmm

coiax committed:
Box: Added drone shell dispenser to Testing Lab

Bemusingly, the merge into this branch removed all our changes, because
when the map changes are this small, it's easier to just scrap them and
make them again.
